pennylane/vqe/vqe.py

```diff
@@ -120,6 +120,32 @@ def __str__(self):
 
         return "\n+ ".join(terms)
 
+    @property
+    def is_diagonal(self):
+        r""" Checks if a Hamiltonian is diagonal in the computational basis.
+
+        Returns:
+            bool: ``True`` if the Hamiltonian is diagonal in the computational basis, ``False`` otherwise
+        """
+
+        diagonals = ["PauliZ", "Identity"]
+
+        non_diagonal_coeffs = []
+        non_diagonal_gates = []
+
+        for i, term in enumerate(self._ops):
+            gates = [term.name] if isinstance(term.name, str) else term.name
+            for j in gates:
+                if j not in diagonals:
+                    if gates not in non_diagonal_gates:
```

Comment:
This double negation is a bit awkward. Can we reformulate? Something like `if gates in diagonal_gates`?